app/main.py

```diff
@@ -1,3 +1,91 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+import operator
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as j_file:
+        info = json.load(j_file)
+
+    fuel_price = info["FUEL_PRICE"]
+
+    customers = []
+    for customer in info["customers"]:
+        customers.append(
+            Customer(
+                customer["name"],
+                customer["product_cart"],
+                customer["location"],
+                customer["money"],
+                customer["car"]
+            )
+        )
+
+    shops = []
+    for shop in info["shops"]:
+        shops.append(
+            Shop(
+                shop["name"],
+                shop["location"],
+                shop["products"]
+            )
+        )
+
+    all_customer_price = []
+    for person in customers:
+        each_customer_price = {}
+        for shop_ in shops:
+            distance = math.sqrt(
+                (person.location[0] - shop_.location[0]) ** 2
+                + (person.location[1] - shop_.location[1]) ** 2
+            )
+            location_cost = ((distance / 100)
+                             * fuel_price
+                             * person.car["fuel_consumption"])
+            product_cost = sum(list(map(operator.mul,
+                                        person.product_cart.values(),
+                                        shop_.products.values())))
+            each_customer_price.update(
+                {shop_.name: round(location_cost * 2 + product_cost, 2)}
+            )
+        all_customer_price.append(each_customer_price)
+
+    for i, d_ in enumerate(all_customer_price):
+        name = customers[i].name
+        f_shop = d_.get("Outskirts Shop")
+        s_shop = d_.get("Shop '24/7'")
+        t_shop = d_.get("Central Shop")
```

Comment:
Retrieve the corresponding shop information using the `get` method from the dictionary